### PR TITLE
Expand on testing requirements

### PR DESCRIPTION
**Pull request by StevenLeighton21**

### CODE REVIEW

File: core-skills/web-application-development-with-react/README.md [hunk 1]
```diff
@@ -118,6 +118,8 @@ The candidate should create a React application which satisfies the following.
   - Candidate should showcase their component to assessor.
   - Candidate should walk assessor through tests which give maintainers confidence
     that the component displays data from an API.
+  - Testing this component should include unit tests that demonstrate the boundary between the view logic and domain logic,
```
Comment (craigjbass) [inline core-skills/web-application-development-with-react/README.md]:
```suggestion
  - Testing this component should include unit tests that demonstrate up to the boundary between the view logic and domain logic,
```

File: core-skills/web-application-development-with-react/README.md [hunk 2]
```diff
@@ -118,6 +118,8 @@ The candidate should create a React application which satisfies the following.
...
+    and acceptance (end to end) testing that demonstrates data retrieved is shown in the UI.
```
Comment (craigjbass) [inline core-skills/web-application-development-with-react/README.md]:
```suggestion
    and acceptance (end to end) testing that demonstrates that all components (which are unit tested independently) interact with each other correctly.
```

I'm not sure we should require it to be UI (I can't think of a better way), but if there is then :+1:

File: core-skills/web-application-development-with-react/README.md [hunk 3]
```diff
@@ -126,6 +128,8 @@ The candidate should create a React application which satisfies the following.
     the form data.
   - Candidate should walk assessor through tests which give maintainers confidence 
     that the component submits the data from the form to the API.
+  - Testing this component should include unit tests that demonstrate the boundary between the view logic and domain logic,
```
Comment (craigjbass) [inline core-skills/web-application-development-with-react/README.md]: dupe?

File: core-skills/web-application-development-with-react/README.md (hunk 2 above)
Comment (StevenLeighton21) [reply core-skills/web-application-development-with-react/README.md]: An end to end automated test is what was _asked_ of me, prompting this update. I don't think that your way is specific enough actually, and a test that shows component interaction but doesn't test UI rendering is moot IMO

File: core-skills/web-application-development-with-react/README.md (hunk 3 above)
Comment (StevenLeighton21) [reply core-skills/web-application-development-with-react/README.md]: Two components, two tests, repeated the requirement to make it clear it was needed in both cases.

File: core-skills/web-application-development-with-react/README.md (hunk 2 above)
Comment (craigjbass) [reply core-skills/web-application-development-with-react/README.md]: I'm concerned that we ask someone to do UI testing, but haven't taught someone WHY we're asking this.

File: core-skills/web-application-development-with-react/README.md (hunk 3 above)
Comment (craigjbass) [reply core-skills/web-application-development-with-react/README.md]: hmm ok :thinking:

File: core-skills/web-application-development-with-react/README.md [hunk 4]
```diff
@@ -126,6 +128,8 @@ The candidate should create a React application which satisfies the following.
...
+    and acceptance (end to end) testing that demonstrates data retrieved is shown in the UI.
```
Comment (craigjbass) [inline core-skills/web-application-development-with-react/README.md]: Mismatch in implementation _oh no!_ duplicate text DUPLICATE TEXT!

**Comment by StevenLeighton21** [comment]
@adrianclay Probably more valuable for the part that is trying to prove _data_ is _shown_, agree there's potentially less value in e2e testing the POST request, as the unit tests should test a reasonable response from the POST API. I feel this could be explicit though, as my original tests were 'spy that the form data was sent' and never verified any sort of response from the API (it didn't exist!).

**Comment by kjdchapman** [comment]
Stale, so closing.
